addon/Feeds/SearchFeed.js

```diff
@@ -23,6 +25,38 @@ module.exports = class SearchFeed extends Feed {
     this.store.dispatch({type: "SEARCH_STATE_UPDATED", data: state});
   }
 
+  /**
+   * getSuggestions - Retrieves search suggestions for a search string (action.data)
+   *                  Dispatches an array of suggestions.
+   */
+  getSuggestions(action) {
+    const browser = getCurrentBrowser();
+    return this.options.searchProvider.asyncGetSuggestions(browser, action.data)
+      .then(suggestions => this.options.send(am.actions.Response("SEARCH_SUGGESTIONS_RESPONSE", suggestions), action.workerId, true))
+      .catch(e => Cu.reportError(e));
```

Comment:
FWIW, I think this may "break" the promise chain since the `.catch()` will swallow any thrown errors and then return a successful promise. Not sure if we want to re-`throw` the error so that any methods calling `getSuggestions()` can handle error cases as appropriate. Or explicitly return an empty array/object in the `catch()` case... ¯\\\_(ツ)\_/¯